odoo/addons/base/models/res_config.py

```diff
@@ -591,7 +591,7 @@ def set_values(self):
         for name, icp in classified['config']:
             field = self._fields[name]
             value = self[name]
-            current_value = current_settings[name]
+            current_value = IrConfigParameter.get_param(icp)
```

Comment:
`get_param` returns a string, meaning `current_value` will be a string, whereas `value` will contain a large variety of types.

Therefore this change will invalidate most checks trying to avoid the set_param call :thinking:.

You should probably update the check below to make sure the comparison still works now that current_value is always a string, and not the same type as `value`

```python
current_value = IrConfigParameter.get_param(icp)
if field.type == 'char':
    ...
elif field.type == 'many2one':
    value = value.id

if current_value == str(value):
    continue
...
```